Tests/Marketplace/search_and_uninstall_pack.py

```diff
@@ -149,14 +166,26 @@ def wait_for_uninstallation_to_complete(client: demisto_client, retries: int = 6
     sleep_duration = 150
     try:
         installed_packs = get_all_installed_packs(client)
+        # Monitoring when uninstall packs don't work
+        installed_packs_amount_history, counter_times_to_repeats = len(installed_packs), 0
+        # new calculation for num of retries
+        retries = math.ceil(len(installed_packs) / 2)
         while len(installed_packs) > 1:
             if retry > retries:
                 raise Exception('Waiting time for packs to be uninstalled has passed, there are still installed '
                                 'packs. Aborting.')
+            if counter_times_to_repeats >= 3:
+                raise Exception(f'Uninstalling packs failed three times. {installed_packs=}')
             logging.info(f'The process of uninstalling all packs is not over! There are still {len(installed_packs)} '
                          f'packs installed. Sleeping for {sleep_duration} seconds.')
             sleep(sleep_duration)
             installed_packs = get_all_installed_packs(client)
+
+            # Monitoring when uninstall packs don't work
+            installed_packs_amount_history, counter_times_to_repeats = \
+                monitoring_uninstall_packs(len(installed_packs), installed_packs_amount_history,
+                                           counter_times_to_repeats)
```

Comment:
I think we can put the logic here, instead of in an external method, whose first return value is its input (unclear)